docs/concepts/smart-contracts.md

```diff
@@ -56,69 +52,73 @@ pub struct HyleOutput {
     pub initial_state: StateDigest,
     pub next_state: StateDigest,
     pub identity: Identity,
-    pub tx_hash: TxHash,
     pub index: BlobIndex,
     pub blobs: Vec<u8>,
+    pub tx_hash: TxHash,
     pub success: bool,
-    pub program_outputs: Vec<u8>,
+    pub tx_ctx: Option<TxContext>, // optional
+    pub registered_contracts: Vec<RegisterContractEffect>,
+    pub program_outputs: Vec<u8>, 
 }
 ```
 
-### Explanation of the ABI fields
-
-#### Version
+### Version
 
 For now, `version` should always be set to 1.
 
-#### Initial state and next state
+### Initial state and next state
 
-Blockchains transactions are fundamentally state transitions. These fields handle state changes securely.
+These fields define state transitions.
 
-- `initial_state`: Matches the onchain `state_digest` before the transaction. The `initial_state` must match the onchain `state_digest`. Otherwise, the state transition is invalid.
+- `initial_state`: must match the onchain `state_digest` before the transaction. If they don't match, the state transition is invalid.
 - `next_state`: Represents the new onchain `state_digest` after the transaction.
 
-Smart contracts can adapt the actual structure of this field. In the future, fees will depend in part on the size of the `state_digest`, so we encourage you to keep it small.
-
-#### Identity
+Future fees may depend on `state_digest` size, so keep it minimal.
```

Comment:
- State digest size will be limited 

- Fees will depend on proof size, bigger states means bigger proof